### PR TITLE
feature: add gen-doc command to auto gen dfget cli docs

### PR DESCRIPTION
**Pull request by allencloud**
Signed-off-by: Allen Sun <allensun.shl@alibaba-inc.com>

### Ⅰ. Describe what this PR did

How to generate user-friendly document is very important work we have to do in Dragonfly project. While I think the best way to make it is to follow the rule **CODE IS DOCUMENT(代码即文档)**. Under such kind of policy, when every update occurs to the CLI-related code, the project has abilities to auto generate the corresponding document. Then the developers can spend less time to construct the document, instead they could have more time to polish the CLI related code when programming which is also reasonable.

We have already supported using `spf13/cobra` to implement COMMAND LINE framework in https://github.com/dragonflyoss/Dragonfly/pull/257. With this work, we can use https://github.com/spf13/cobra/blob/master/doc/md_docs.md to add another command to generate all the CLI document.

After adding the command which is used to generate CLI document, we can make @pouchrobot to automatically generate doc every day when it has update to the cli part. This is quite similar to [PouchContainer](https://github.com/alibaba/pouch).

### Ⅱ. Does this pull request fix one issue?

fix https://github.com/dragonflyoss/Dragonfly/issues/306

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

no need 

### Ⅳ. Describe how to verify it
none

### Ⅴ. Special notes for reviews
none

### CODE REVIEW

**Comment by allencloud** [comment]
After PR https://github.com/pouchcontainer/pouchrobot/pull/100 is merged, we could make @pouchrobot to support auto generate cli documentation.

**Comment by Starnop** [comment]
Please rebase the master branch after [PR](https://github.com/dragonflyoss/Dragonfly/pull/305) merged.

**Comment by allencloud** [comment]
PTAL @Starnop

File: cmd/dfget/app/gen_doc.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		SilenceErrors: true,
+		SilenceUsage:  true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return genDocCommand.runGenDoc(args)
+		},
+	}
+	rootCmd.AddCommand(genDocCommand.cmd)
+}
+
+func (g *GenDocCommand) runGenDoc(args []string) error {
```
Comment (Starnop) [inline cmd/dfget/app/gen_doc.go]: Could you please implement the related part of `dfdaemon` by he way?

Comment (allencloud) [reply cmd/dfget/app/gen_doc.go]: Yes, I will do that in a separate PR.

File: docs/cli_reference/dfget.md [hunk 2]
```diff
@@ -1,83 +1,43 @@
...
-  --version, -v         version
-  --node NODE, -n NODE  specify nodes
-  --console             show log on console
-  --header HEADER       http header, e.g. --header="Accept: *" --header="Host:
-                        abc"
-  --dfdaemon            caller is from df-daemon
+dfget [flags]
 ```
 
-## Files
```
[the remark this answers is not in the document]
Comment (allencloud) [reply docs/cli_reference/dfget.md]: Yes, we can split that from the dfget.md.

**Comment by allencloud** [comment]
I am wondering if we could add a new subcommand of `dfget pull` for dfget. Then it would be much clearer for the usage of dfget. @lowzj @Starnop

**Pull request merged**